src/Raven.Server/Web/System/SetupHandler.cs

```diff
@@ -0,0 +1,383 @@
+                HttpContext.Response.ContentType = "binary/octet-stream";
+
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+                HttpContext.Response.Body.Write(zip, 0, zip.Length);
+            }
+        }
+
+        [RavenAction("/setup/letsencrypt/agreement", "GET", AuthorizationStatus.UnauthenticatedClients)]
+        public async Task SetupAgreement()
+        {
+            AssertOnlyInSetupMode();
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var setupInfoJson = context.ReadForMemory(RequestBodyStream(), "claim-domain"))
+            {
+                var setupInfo = JsonDeserializationServer.SecuredSetupInfo(setupInfoJson);
+
+                var uri = await ServerStore.SetupManager.LetsEncryptAgreement(setupInfo.Email);
+
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    writer.WriteStartObject();
+                    writer.WritePropertyName("Uri");
+                    writer.WriteString(uri.AbsolutePath);
+                    writer.WriteEndObject();
+                }
+            }
+
+            HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
+        }
+
+        [RavenAction("/setup/letsencrypt", "POST", AuthorizationStatus.UnauthenticatedClients)]
+        public Task SetupLetsEncrypt()
+        {
+            AssertOnlyInSetupMode();
+
+            var stream = TryGetRequestFromStream("Options") ?? RequestBodyStream();
+
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var setupInfoJson = context.ReadForMemory(stream, "setup-lets-encrypt"))
+            {
+                var setupInfo = JsonDeserializationServer.SecuredSetupInfo(setupInfoJson);
+
+                var operationCancelToken = new OperationCancelToken(ServerStore.ServerShutdown);
+                var operationId = ServerStore.Operations.GetNextOperationId();
+
+                ServerStore.Operations.AddOperation(
+                    null,
+                    "Setting up RavenDB with a Let's Encrypt certificate",
+                    Documents.Operations.Operations.OperationType.SetupLetsEncrypt,
+                    progress => ServerStore.SetupManager.FetchCertificateTask(progress, operationCancelToken.Token, setupInfo),
+                    operationId, operationCancelToken);
+
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    writer.WriteOperationId(context, operationId);
+                }
+
+                // Here we need to take the jsons, create files and zip them all together
+                // Need to make this an async operation, just like when sending a certificate for the studio... to know that we're done and sending a file
```

Comment:
You cannot do this properly. You have an ongoing operation and the request has completed.
You need a _separate_ request to get the data.